compiler/rustc_trait_selection/src/traits/specialize/mod.rs

```diff
@@ -200,36 +202,37 @@ fn fulfill_implication<'tcx>(
         return Err(());
     };
 
+    // Needs to be `in_snapshot` because `fulfill_implication` is used to rebase substitutions
```

Comment:
```suggestion
    // Needs to be `in_snapshot` because this method is used to rebase substitutions
```
i was confused for a few seconds why `fulfill_implication` is relevant here :grin: